src/pages/FoodItem/FoodItemPage.tsx

```
@@ -34,7 +34,30 @@ export default function FoodItemPage() {
     document.title = foodItem ? foodItem.name : "Not Found";
   });
```

Don't get from the title. The source of truth should be a single point. Not getting data from here and there and checking them. The source of truth should be the database. If it's not there it means it's not found

---

And useEffect should always have a dependency array

---

Removed useEffect.
What do you mean don't get from the title? I am setting the title's name. 

As for getting the data, what I did is: 
```
const { foodItemName } = useParams(); //gets the foodItemName from the URL
const foodItem = mockDatabase.find(
    (item) => _.kebabCase(item.name) === foodItemName
) as FoodItem;
```
Then the line that you quoted:
```
document.title = foodItem.name;
```

Previously, I didn't add the line `as FoodItem`, so I had "might be undefined" error. And I used useEffect, which wasn't necessary.

An issue I foresee is that if the `mockDatabase.find()` returns undefined, what do I do?
Because when I try 
`if (foodItem === undefined) return <NotFound/>;`
All of the hooks below tells me that I can't call useHook conditionally. But those hooks rely on foodItem.

---

Try not to update the document directly. People usually use this package to update the elements in the <head>https://www.npmjs.com/package/react-helmet-async

---

You get the conditional hooks issue because you are returning before the declarations of all the hooks, so some hooks are run while some hooks aren't when the condition is met. You can try to move the return value below all the hook declarations. You can also have a placeholder value when it's not found. for eg, foodItem ?? {name: '', price: 0, etc}. Then just return the NotFound error, so the placeholder won't ever be shown

---

My point was that you should update all the data from 1 single source of truth, which is the database. Don't use the database to update something and then use that to update something else. Have all references to the data linked back to the source of truth(DB). 
Otherwise if the DB data is changed in some way, you may forget to update the intermediary code and cause the displayed info to be wrong

---

> Try not to update the document directly. People usually use this package to update the elements in the https://www.npmjs.com/package/react-helmet-async

I shall not change the document.title then.


> You get the conditional hooks issue because you are returning before the declarations of all the hooks, so some hooks are run while some hooks aren't when the condition is met. You can try to move the return value below all the hook declarations. You can also have a placeholder value when it's not found. for eg, foodItem ?? {name: '', price: 0, etc}. Then just return the NotFound error, so the placeholder won't ever be shown

Fixed this by using the method you've described.